crates/apub/src/objects/private_message.rs

```diff
@@ -130,9 +136,18 @@ impl Object for ApubPrivateMessage {
     let recipient = note.to[0].dereference(context).await?;
     PersonBlock::read(&mut context.pool(), recipient.id, creator.id).await?;
 
+    // If its a local user, check that they can receive private messages
+    if recipient.local {
+      if let Ok(recipient_local_user) =
+        LocalUserView::read_person(&mut context.pool(), recipient.id).await
+      {
+        check_private_messages_enabled(&recipient_local_user)?;
+      }
```

Comment:
The check for apub is done here, so it is working, its just not a publicized setting, so like blocks, senders don't know.